pkg/kube/secretcontroller/secretcontroller.go

```diff
@@ -249,13 +254,16 @@ func (c *Controller) close() {
 
 func (c *Controller) hasSynced() bool {
 	if !c.initialSync.Load() {
+		log.Debug("secret controller did not syncup secrets presented at startup")
 		// we haven't finished processing the secrets that were present at startup
 		return false
 	}
 	c.cs.RLock()
-	defer c.cs.RUnlock()
-	for _, cluster := range c.cs.remoteClusters {
+	rc := c.cs.remoteClusters
+	c.cs.RUnlock()
+	for _, cluster := range rc {
 		if !cluster.HasSynced() {
+			log.Debugf("remote cluster %s secrets have not been synced up yet", cluster.secretName)
```

Comment:
Not sure if `secrets have not been` is accurate. This will check if any of the informers registered for that cluster's kube client have synced.

Reply:
Agree. I will correct